doc/developers-guide/Hooks-and-handlers.md

```diff
@@ -66,23 +66,13 @@ If you haven't encountered the term _fold_ before, think of it as _reduce_ (like
 
 ### Using accumulators
 
-MongooseIM uses a dedicated data structure to accumulate results (see ["Accumulators"](accumulators.md)).
-This data structure is implemented in the `mongoose_acc` module, which has a map-like interface: it has `get/2`, `get/3`, `put/3` etc.
-It is instantiated with an incoming stanza, passed along throughout the processing chain, supplied to and returned from hook calls, and terminated when stanza is leaving MongooseIM.
-If hook handlers are supposed to return some value they put it into the accumulator.
+MongooseIM uses a dedicated data structure to accumulate data related to stanza processing (see ["Accumulators"](accumulators.md)).
+It is instantiated with an incoming stanza, passed along throughout the processing chain, supplied to and returned from certain hook calls, and terminated when stanza is leaving MongooseIM.
 
-The right way to use hooks is therefore:
-
-* create a handler which takes and returns an accumulator
-* take an accumulator if available, or instantiate a new one
-* call run_fold giving your acc as the accumulator, plus some extra arguments as needed
-* take return value from the acc the hook call returned
-* pass the modified accumulator on
-
-Handlers should store their return values in the accumulator; there are three ways to do it:
-* if it is a one-off value whch doesn't need to be passed on along with the accumulator (can be overwritten any time), use `mongoose_acc:put(result, Value, Acc)`
-* if the value is to be passed on to be reused within the user's session use `mongoose_acc:put(Key, Value, Acc)`
-* if the value should be passed on to the recipient's session, pubsub node etc. use `mongoose_acc:add_prop(Key, Value, Acc)`
+If a Mongoose accumulator is provided to a hook, handlers should store their return values in one of 3 ways:
+* If it is a one-off value whch doesn't need to be passed on along with the accumulator (can be overwritten any time), use `mongoose_acc:set(hook, result, Value, Acc)`.
```

Comment:
If it is a one-off value which doesn't need to be passed on along with the accumulator (can be overwritten any time), use `mongoose_acc:set(hook, result, Value, Acc)`.